lib/Launcher.js

```diff
@@ -86,12 +89,16 @@ class Launcher {
       chromeProcess.stderr.pipe(process.stderr);
     }
 
-    const waitForChromeToClose = new Promise(fulfill => {
+    const waitForChromeToClose = new Promise((fulfill, reject) => {
```

Comment:
```js
const waitForCloseEvent = () => new Promise(resolve => chromeProcess.once('close', resolve);
const waitForChromeToClose = (async () => {
  await waitForCloseEvent();
  if (temporaryUserDateDir) {
    try {
      await removeFolderAsync(temporaryUserDataDir);
    } catch(e) {
       console.error(e); // are you sure?
    }
  }
})();

Reply:
@aslushnikov I disagree, new Promise should not be mixed with x.then.

Reply:
@graingert I might be misunderstanding something. Lets split this discussion out of this thread: would you mind sending a separate PR that does changes that you find important and explains why are they necessary?